src/Codeception/Lib/Connector/Phalcon/MemorySession34x.php

```diff
@@ -0,0 +1,317 @@
+<?php
+namespace Codeception\Lib\Connector\Phalcon;
+
+use Phalcon\Session\AdapterInterface;
+
+class MemorySession34x implements AdapterInterface
+{
+    /**
+     * @var string
+     */
+    protected $sessionId;
+
+    /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @var bool
+     */
+    protected $started = false;
+
+    /**
+     * @var array
+     */
+    protected $memory = [];
+
+    /**
+     * @var array
+     */
+    protected $options = [];
+
+    /**
+     * MemorySession constructor.
+     * @param array|null $options
+     */
+    public function __construct(array $options = null)
+    {
+        $this->sessionId = $this->generateId();
+
+        if (is_array($options)) {
+            $this->setOptions($options);
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function start()
+    {
+        if ($this->status() !== PHP_SESSION_ACTIVE) {
+            $this->memory = [];
+            $this->started = true;
+
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @param array $options
+     */
+    public function setOptions(array $options)
+    {
+        if (isset($options['uniqueId'])) {
+            $this->sessionId = $options['uniqueId'];
+        }
+
+        $this->options = $options;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return array
+     */
+    public function getOptions(): array
```

Comment:
@sergeyklay but the branch 3.4.x 

```
	/**
	 * Get internal options
	 */
	public function getOptions() -> array;
```
compiles to
```
public function getOptions(): array
```
that's the whole reason why I wrote this patch my local codeception was not working anymore :)

please correct me if I am wrong.

Reply:
Let me explain. Phalcon 3.4.x as well as current Codeception _must_ works on PHP 5.x. This fact confirmed by tests in the corresponding repositories. Such a function definition isn't compatible with PHP 5 syntax:
```
public function getOptions(): array
```

Reply:
And yes, this is not PHP:
```zep
public function getOptions() -> array;
```
This is Zephir.

Reply:
@sergeyklay hmm well I know that but than it's incompatible with your implementation. so I need to build an extra check for the version > 7. and rename it. Because the only reason why i need this adapter is because phalcon breaks codeception in version 7.3 because phalcon changed the interfaces / the compiler now generates strict signature interface contracts for version 7+.

So I guess I need to check for the php >=7 ∩ Phalcon 3.4.x? with 7.2 and phalcon 3.3.x compile with zephir 10.x i didn't have the problem.

but php 7.3 + zephir 11.x + phalcon 3.4.x breaks it and 3.3.x or 4.0.x have different issues in this combination one time I need psr-ext and the other time I get a list of all the internal signature contracts that are wrong. Besides my gcc downgrades needed to compile it. (rust needs a newer gcc)

does this make sense to you? 

tl;dr; 
if I just do what you ask php 5 will work but 7.3 with phalcon 3.4.x compiled with zephir 11.x will break

that's it in a nutshell.

Reply:
Backward incompatible changes are in Zephir 0.11.x and Phalcon 4.x. So you may need to compile Phalcon 3.4.x using Zephir 0.10.x. Also take a look at https://stackoverflow.com/a/54066201/1661465

Reply:
@sergeyklay fair enough, I use arch and I compile things per hand. So good for rpm and deb architectures. https://aur.archlinux.org/packages/php-phalcon/ currently defines it to 7.2 so also no luck there.

So I will close this PR as invalid, since I already fixed it in my fork and I don't have to care about compatibility. 

However it would be nice if you mention the zephir version locking with v3 of phalcon in the phalcon compilation readme guide. :) 

my mistake and sorry for wasting all of your time :)